expression/builtin_compare_vec.go

```diff
@@ -801,11 +801,50 @@ func (b *builtinLeastTimeSig) vecEvalString(input *chunk.Chunk, result *chunk.Co
 }
 
 func (b *builtinNERealSig) vectorized() bool {
-	return false
+	return true
 }
 
 func (b *builtinNERealSig) vecEvalInt(input *chunk.Chunk, result *chunk.Column) error {
-	return errors.Errorf("not implemented")
+	n := input.NumRows()
+	buf0, err := b.bufAllocator.get(types.ETReal, n)
+	if err != nil {
+		return err
+	}
+	defer b.bufAllocator.put(buf0)
+	if err := b.args[0].VecEvalReal(b.ctx, input, buf0); err != nil {
+		return err
+	}
+	buf1, err := b.bufAllocator.get(types.ETReal, n)
+	if err != nil {
+		return err
+	}
+	defer b.bufAllocator.put(buf1)
+	if err := b.args[1].VecEvalReal(b.ctx, input, buf1); err != nil {
+		return err
+	}
+	result.ResizeInt64(n, false)
+	result.MergeNulls(buf0, buf1)
+	i64s := result.Int64s()
+	x := buf0.Float64s()
+	y := buf1.Float64s()
+	for i := 0; i < n; i++ {
+		isNull0 := buf0.IsNull(i)
+		isNull1 := buf1.IsNull(i)
+		if isNull0 || isNull1 {
+			if compareNull(isNull0, isNull1) == 0 {
+				i64s[i] = 0
+			} else {
+				i64s[i] = 1
+			}
+		} else {
+			if x[i] == y[i] {
```

Comment:
I think we'd better use `types.CompareFloat64` here. Because we may refine the implementation of `double0 == double1` in it.

Reply:
ok